### PR TITLE
Email mapper

### LINKED ISSUE #26: Implement Email mapper
Reference: https://oozie.apache.org/docs/5.1.0/DG_EmailActionExtension.html

### PR DESCRIPTION
**Pull request by sprzedwojski**
Make sure you have checked _all_ steps below and that above description is correct.

### JIRA
- [x] My PR addresses the following [Issue](https://github.com/GoogleCloudPlatform/oozie-to-airflow/issues/26)
issue and references it in commit message.

### Tests
- [x] My PR has unit tests testing the functionality (or I explained why it needs no tests)

### Commits
- [x] My commits reference the issue in the message using: Fixes #XXX /Closes #XXX or similar.
Closes #26

### Documentation
- [x] My PR adds documentation that describes how to use it (or I explained why it needs no docs)

### CODE REVIEW

**Comment by sprzedwojski** [comment]
Instructions on how the MailHog test SMTP server is being run: https://hackmd.io/@spr/HJFYU-qlr

File: pylintrc [hunk 1]
```diff
@@ -426,6 +426,8 @@ good-names=i,
            k,
            ex,
            Run,
+           to,
```
Comment (potiuk) [inline pylintrc]: I think we should seriously consider email_to and email_cc instead. This way we would not have to make exclusions here. I think exclusions in pylintrc should be "generic" - i.e. names that are universally used as short names in well-known way (indexes, exceptions etc.). "to" and "cc" are very specific to email case.

File: README.md [hunk 2]
```diff
@@ -712,3 +716,77 @@ Connection configuration options are not supported.
...
+
+*Solution:* Implement in O2A a mechanism to extract a file from HDFS inside Airflow.
+
+**2. `<content_type>` tag is not supported**
+
+From Oozie docs:
+> From uri:oozie:email-action:0.2 one can also specify mail content type as
+<content_type>text/html</content_type>. “text/plain” is default.
+
+Unfortunately, currently the `EmailOperator` only accepts the `mime_subtype` parameter.
```
Comment (potiuk) [inline README.md]: Nice that we have it written down.

File: README.md [hunk 3]
```diff
@@ -712,3 +716,77 @@ Connection configuration options are not supported.
...
+
+*Solution:* Modify the Airflow's `EmailOperator` to support more
+content types.
+
+**3. `cc` and `bcc` fields are not templated in EmailOperator**
+
+Only the 'to', 'subject' and 'html_content' fields in EmailOperator are templated.
+In practice this covers all fields of an Oozie email action node apart from `cc` and `bcc`.
+
+Therefore if there is an EL function in the action node in either of these two fields
```
Comment (potiuk) [inline README.md]: Good to know those limitations!

File: README.md [hunk 4]
```diff
@@ -92,6 +92,10 @@ If you want to contribute to the project, please take a look at [CONTRIBUTING.md
   - [Hive/Hive2 Example](#hivehive2-example)
     - [Output](#output-12)
     - [Current limitations](#current-limitations-12)
+  - [Email Example](#email-example)
+    - [Output](#output-13)
+    - [Prerequisites](#prerequisites)
+    - [Known limitations](#known-limitations)
```
Comment (potiuk) [inline README.md]: I think we should decide wether to use "Current" or "Known" and use it consistently. I agree "Known limitations" is better but then we should likely make another PR fixing the "Current limitations" names in other places.

File: pylintrc (hunk 1 above)
Comment (sprzedwojski) [reply pylintrc]: Good idea. I was thinking on how to replace `to` and `cc` and had no ideas, but `email_to` and `email_cc` do look good, thanks.

File: README.md (hunk 4 above)
Comment (sprzedwojski) [reply README.md]: Yes, I also had that feeling. I'll quickly prepare a PR.

Comment (sprzedwojski) [reply README.md]: https://github.com/GoogleCloudPlatform/oozie-to-airflow/pull/291

File: pylintrc (hunk 1 above)
Comment (sprzedwojski) [reply pylintrc]: In the end I opted for `to_addr`, `cc_addr` and `bcc_addr` (the last one for consistency).
e68d601b5acd9b1773c218219eeb7fb33e068e6c
